docs/source/concepts/IndexProperty.rst

```diff
@@ -0,0 +1,101 @@
+.. _IndexProperty:
+
+Index Property
+==============
+
+.. contents::
+  :local:
+
+What is an IndexProperty?
+------------------------
+
+The ``IndexProperty`` in Mantid provides a consistent interface for algorithms 
```

Comment:
I don't think this section properly justifies why future developers should not just use `ArrayProperty<int>`. Can you add more on this and explicitly say that in future stuffing indexes into `ArrayProperty<int>` will be deprecated.